### PR TITLE
Correct default value comments

### PR DESCRIPTION
**Pull request by saicheems**
Redoes some parts of DefaultString to make it more accessible in
snippetgen. Also delegates line endings and comments to snippetgen so
none of that work is done in the contexts.

### CODE REVIEW

**Comment by saicheems** [comment]
OK! This is a surprising amount of work for a simple thing 😞 

This change refactors a good amount of DefaultString and defaultValue so they don't print semicolons, commas, comments, and a bunch of formatting that should really be at the snippet layer. For all of this work, it fixes the special comments after default fields so they don't absorb semicolons at the end of lines and cause compiler errors.

This is ready for review.

**Comment by saicheems** [comment]
I've only added the compute baseline for ease of review. As far as I can tell, it's the only type that in the discovery samples that has an "eg. " style comment.

**Comment by pongad** [comment]
I admit that I'm a little upset so much change is necessary for this, but I don't see any other way out either. LGTM

File: src/main/java/com/google/api/codegen/discovery/DefaultString.java [hunk 1]
```diff
@@ -75,25 +75,25 @@ public String getComment() {
...
+  public static String getSample(String apiName, String fieldName, String pattern) {
     String sample = null;
+    if (pattern != null) {
+      // If the pattern has a specially-recognized sample, use the sample.
+      sample = SAMPLE_STRINGS.get(SampleKey.create(apiName, fieldName, pattern));
+    }
+    return sample == null ? "" : sample;
+  }
 
+  public static String getDefault(String fieldName, String pattern) {
```
Comment (vchudnov-g) [inline src/main/java/com/google/api/codegen/discovery/DefaultString.java]: nit: getSample() vs getDefault() seems a bit confusing. I would s/getDefault/getPlaceholder/

File: src/main/java/com/google/api/codegen/go/GoDiscoveryContext.java [hunk 2]
```diff
@@ -64,17 +64,12 @@ public String typeDefaultValue(Type type, Field field) {
       return DEFAULT_VALUES.get(field.getKind());
     }
     if (field.getKind() == Field.Kind.TYPE_STRING || field.getKind() == Field.Kind.TYPE_ENUM) {
-      return getDefaultString(type, field);
+      return getDefaultString(type, field).getDeclare();
```
Comment (vchudnov-g) [inline src/main/java/com/google/api/codegen/go/GoDiscoveryContext.java]: nit: Is getDeclare actually a getDefine? You're defining the value, not declaring the type, IIC.

(We can fix the names later if need. I'm probably misunderstanding.)

**Comment by vchudnov-g** [comment]
Some nits on naming, which we could defer.

If tests pass, LGTM.

File: src/main/java/com/google/api/codegen/discovery/DefaultString.java (hunk 1 above)
Comment (saicheems) [inline src/main/java/com/google/api/codegen/discovery/DefaultString.java]: Done

File: src/main/java/com/google/api/codegen/go/GoDiscoveryContext.java (hunk 2 above)
Comment (saicheems) [inline src/main/java/com/google/api/codegen/go/GoDiscoveryContext.java]: Done

**Pull request merged**